### PR TITLE
Update libwebrtc version, bump dependencies to latest, refactor to use latest gradle

### PR DESCRIPTION
**Pull request by sirknightj**
## What was changed?

Bumped dependencies, and removed some unused ones.
* com.android.tools.build:gradle:3.5.3 -> 7.3.3
* org.webrtc:google-webrtc:jar:1.0.32006 --> M114
* com.amazonaws:aws-android-sdk-kinesisvideo, com.amazonaws:aws-android-sdk-kinesisvideo-signaling, com.amazonaws:aws-android-sdk-mobile-client, com.amazonaws:aws-android-sdk-auth-userpools, com.amazonaws:aws-android-sdk-kinesisvideo-webrtcstorage 2.68.0 -> 2.73.0
* aws-android-sdk-auth-ui-debug.aar -> com.amazonaws:aws-android-sdk-auth-ui:2.73.0
* org.awaitility:awaitility:3.0.0 -> 4.2.0
* com.google.code.gson:gson:2.8.6 -> 2.8.9
* org.slf4j:slf4j-api:1.7.29 -> x
* com.github.tony19:logback-android:2.0.0 -> x
* commons-codec:commons-codec:1.9 -> x
* org.osgi:org.osgi.framework:1.9.0 -> x
* androidx.appcompat:appcompat:1.1.0 -> 1.6.1
* androidx.constraintlayout:constraintlayout:1.1.3 -> 2.1.4
* com.google.android.material:material:1.0.0 -> 1.9.0
* androidx.multidex:multidex:2.0.1 -> x
* com.dicedmelon.gradle:jacoco-android:0.1.4 -> x

Also, fixed a small UI issue where clicking the triangle back button while on the Channel configuration screen would take you to the empty sign-in activity.

## Why was it changed?
* Google is no longer vending out prebuilt binaries of their webrtc library, and have offloaded that responsibility to the consumers. org.webrtc:google-webrtc:jar:1.0.32006 was published on Aug 27, 2020, which is several years old at this point. The last version that Google released contains a dependency with a vulnerability according to https://support.google.com/faqs/answer/12577537?hl=en, which mentions that it has been fixed as of M102. I have built M114 (https://chromiumdash.appspot.com/branches) following https://webrtc.github.io/webrtc-org/native-code/android/ and have included it in the `libs` folder. The latest webrtc library was not compatible with the version of gradle that was previously used, so that needed bumping. Gradle has changed significantly since the version we were using, so the project structure needed a refactor as well.

## Testing
* Tested android as both master and viewer with the JS sample (https://awslabs.github.io/amazon-kinesis-video-streams-webrtc-sdk-js/examples/index.html)
* Verified the sample application is able to connect using the WebRTC ingestion feature in us-west-2.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: app/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java [hunk 1]
```diff
@@ -185,7 +187,7 @@ private void initWsConnection() {
         final String masterEndpoint = mWssEndpoint + "?" + Constants.CHANNEL_ARN_QUERY_PARAM + "=" + mChannelArn;
 
         // See https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-websocket-apis-1.html
```
Comment (unicornss) [inline app/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java:189]: https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-websocket-apis.html

File: app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar', '*.aar'])
+
+    def aws_version = '2.73.0'
+    implementation("com.amazonaws:aws-android-sdk-kinesisvideo:$aws_version@aar") { transitive = true }
+    implementation("com.amazonaws:aws-android-sdk-kinesisvideo-signaling:$aws_version@aar") { transitive = true }
+    implementation("com.amazonaws:aws-android-sdk-mobile-client:$aws_version@aar") { transitive = true }
+    implementation("com.amazonaws:aws-android-sdk-auth-userpools:$aws_version@aar") { transitive = true }
+    implementation("com.amazonaws:aws-android-sdk-auth-ui:$aws_version@aar") { transitive = true }
+    implementation("com.amazonaws:aws-android-sdk-kinesisvideo-webrtcstorage:$aws_version@aar") { transitive = true }
```
Comment (unicornss) [inline app/build.gradle:40]: /nit can we group kinesisvideos together

Comment (sirknightj) [reply app/build.gradle:40]: Done

File: app/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java (hunk 1 above)
Comment (sirknightj) [reply app/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java:189]: I linked ConnectAsViewer above the viewer URL and ConnectAsMaster above the master URL

**Review by unicornss — APPROVED** [review]

**Review by unicornss — APPROVED** [review]

**Review by unicornss — APPROVED** [review]

**Review by unicornss — APPROVED** [review]

**Review by unicornss — APPROVED** [review]

**Review by hassanctech — APPROVED** [review]
lgtm

**Comment by sirknightj** [comment]
Added an error message for when `libwebrtc` is not detected (rather than letting it error out with "class X not found"). Here's what you'll see/run into when building the project without it.

image

**Pull request merged**